src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -0,0 +1,417 @@
+  (λ hf, le_antisymm (real.Inf_le _ bounds_bdd_below
+    ⟨ge_of_eq rfl, λ _, le_of_eq (by { rw [zero_mul, hf], exact norm_zero })⟩)
+    (op_norm_nonneg _))
+
+variables {f g}
+
+/-- The identity as a continuous normed group hom. -/
+@[simps]
+def id : normed_group_hom V V :=
+(add_monoid_hom.id V).mk_continuous 1 (by simp [le_refl])
+
+/-- The norm of the identity is at most `1`. It is in fact `1`, except when the space is trivial
+where it is `0`. It means that one can not do better than an inequality in general. -/
+lemma norm_id_le : ∥(id : normed_group_hom V V)∥ ≤ 1 :=
+op_norm_le_bound _ zero_le_one (λx, by simp)
+
+/-- If a space is non-trivial, then the norm of the identity equals `1`. -/
+lemma norm_id [nontrivial V] : ∥(id : normed_group_hom V V)∥ = 1 :=
+le_antisymm norm_id_le $ let ⟨x, hx⟩ := exists_ne (0 : V) in
+have _ := (id : normed_group_hom V V).ratio_le_op_norm x,
+by rwa [id_apply, div_self (ne_of_gt $ norm_pos_iff.2 hx)] at this
+
+/-- Opposite of a normed group hom. -/
+instance : has_neg (normed_group_hom V₁ V₂) :=
+⟨λ f, (-f.to_add_monoid_hom).mk_continuous (∥f∥) (λ v, by simp [le_op_norm f v])⟩
+
+/-- Subtraction of normed group homs. -/
+instance : has_sub (normed_group_hom V₁ V₂) :=
+⟨λ f g,
+{ bound' :=
+  begin
+    simp only [add_monoid_hom.sub_apply, add_monoid_hom.to_fun_eq_coe, sub_eq_add_neg],
+    exact (f + -g).bound'
+  end,
+  .. (f.to_add_monoid_hom - g.to_add_monoid_hom) }⟩
+
+@[simp] lemma coe_zero : ⇑(0 : normed_group_hom V₁ V₂) = 0 := rfl
+
+@[simp] lemma coe_neg (f : normed_group_hom V₁ V₂) : ⇑(-f) = -f := rfl
+
+@[simp] lemma coe_add (f g : normed_group_hom V₁ V₂) : ⇑(f + g) = f + g := rfl
+
+@[simp] lemma coe_sub (f g : normed_group_hom V₁ V₂) : ⇑(f - g) = f - g := rfl
+
+lemma op_norm_neg (f : normed_group_hom V₁ V₂) : ∥-f∥ = ∥f∥ :=
+by simp only [norm_def, coe_neg, norm_neg, pi.neg_apply]
+
+/-- Homs between two given normed groups form a commutative additive group. -/
+instance : add_comm_group (normed_group_hom V₁ V₂) :=
+by refine_struct
+{ .. normed_group_hom.has_add, .. normed_group_hom.has_zero,
+  .. normed_group_hom.has_neg, ..normed_group_hom.has_sub };
+{ intros, ext, simp [add_assoc, add_comm, add_left_comm, sub_eq_add_neg] }
+
+/-- Normed group homomorphisms themselves form a normed group with respect to
+    the operator norm. -/
+instance to_normed_group : normed_group (normed_group_hom V₁ V₂) :=
+normed_group.of_core _ ⟨op_norm_zero_iff, op_norm_add_le, op_norm_neg⟩
+
+/-- Coercion of a `normed_group_hom` is an `add_monoid_hom`. Similar to `add_monoid_hom.coe_fn` -/
+@[simps]
+def coe_fn_add_hom : normed_group_hom V₁ V₂ →+ (V₁ → V₂) :=
+{ to_fun := coe_fn, map_zero' := coe_zero, map_add' := coe_add}
+
+@[simp] lemma coe_sum {ι : Type*} (s : finset ι) (f : ι → normed_group_hom V₁ V₂) :
+  ⇑(∑ i in s, f i) = ∑ i in s, (f i) :=
+(coe_fn_add_hom : _ →+ (V₁ → V₂)).map_sum f s
+
+lemma sum_apply {ι : Type*} (s : finset ι) (f : ι → normed_group_hom V₁ V₂) (v : V₁) :
```

Comment:
Given you don't have `add_apply`, `sub_apply` etc, I'd argue you should either add those lemmas too, or remove this one.

Reply:
I have added the other ones.

Reply:
Thanks